### ISSUE TITLE
DevContainer Feature

### ISSUE THREAD

**Issue by jarrodcolburn** [neovim/neovim#22569]
### Problem

Currently on [DevContainer Features](https://containers.dev/features) there are two different installations for NeoVim, via `apt` and `homebrew` provided by a third party.

![Screenshot 2023-03-07 21 17 39](https://user-images.githubusercontent.com/16673615/223611017-403df970-684e-427c-8285-ca6e352b026c.png)

However, the `apt` installation is currently not using the `ppa` resulting in installation of legacy NeoVim `0.6` [issue](https://github.com/devcontainers-contrib/features/issues/328)

And the `homebrew` installation requires a lengthy installation of homebrew prior.

Currently, NeoVim maintainers are not distributing a DevContainer.

## Add DevContainer Feature as distribution method.

NeoVim is already doing an amazing, amazing job at linux distribution. With `apt`,`yum`,`appimage`,`flatpack`,.... the list goes on.... so I hate to suggest yet another way to release Neovim. But *initially a DevContainer feature that installs the `apt` version would be appreciated, (as `debian` is the testing version).

Although,  distribution as `devcontainers` might seem ironic at first, as devcontainers are tightly linked to `VS Code` and `GitHub Codespaces`, there are many people who want to run...
```bash
# using GitHub's gh cli
gh codespace ssh
```
...to get an ssh shell into their Codespace, why? To start using their beloved `NeoVim`! 😃

Speaking of GitHub (Codespaces, since it runs on devcontainers), I feel that these distribution model would strengthen NeoVim for those who want to use Copilot too.

*In the future, one that detects the linux distro, and uses the appropriate package manager (`apt`,yum`,`apk`,...) might be nice.

**Comment by dundargoc** [comment]
Neovim generally doesn't take distribution or packaging roles like that as it tends to not be sustainable or maintainable in the long run. Instead we rely on amazing contributors who help us with packaging in their own distribution. It's also better as these people are already familiar with their own distribution and package manager.

That said, neovim will happily help and unblock anyone interested in getting this to work.

**Comment by clason** [comment]
> With apt,yum,appimage,flatpack,.... the list goes on....

None of which are maintained by the Neovim org (except appimage, which is a flat file and not a distribution platform). You should raise this issue (up-to-date Neovim, e.g., using our appimage) with the DevContainer project.

As @dundargoc said, feel free to open a new issue if there are specific problems with our build or release workflow that prevent that.

**Comment by jarrodcolburn** [comment]
ahhh, yea. Gotcha, dang, I feel silly opening this in hindsight. But, I sincerely appreciate you two explaining things to me. 

I am a huge fan of `appimage`! If that's the one NeoVim Org is maintaining, that's a great starting point for me, thanks.

**Comment by justinmk** [comment]
Related: 

- https://github.com/neovim/neovim/issues/21635
